src/Workspaces/Core/Portable/Execution/Serialization/Serializer_Asset.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    /// <summary>
+    /// serialize and deserialize objects to stream.
+    /// some of these could be moved into actual object, but putting everything here is a bit easier to find I believe.
+    /// </summary>
+    internal partial class Serializer
+    {
+        public void Serialize(SolutionChecksumObjectInfo info, ObjectWriter writer, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            Serialize(info.Id, writer, cancellationToken);
+
+            // TODO: figure out a way to send version info over as well.
+            //       right now, version get updated automatically, so 2 can't be exactly match
+            // info.Version.WriteTo(writer);
+            writer.WriteString(info.FilePath);
+        }
+
+        private SolutionChecksumObjectInfo DeserializeSolutionSnapshotInfo(ObjectReader reader, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var solutionId = DeserializeSolutionId(reader, cancellationToken);
+            // var version = VersionStamp.ReadFrom(reader);
+            var filePath = reader.ReadString();
+
+            return new SolutionChecksumObjectInfo(solutionId, VersionStamp.Create(), filePath);
+        }
+
+        public void Serialize(ProjectChecksumObjectInfo info, ObjectWriter writer, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            Serialize(info.Id, writer, cancellationToken);
+
+            // TODO: figure out a way to send version info over as well
```

Comment:
we already have those. it is not because we can't send them over, it is because workspace mutate those. I have work item tracking for supporting versioning.